client/my-sites/themes/theme-upload/index.jsx

```diff
@@ -257,27 +305,65 @@ class Upload extends React.Component {
 	}
 
 	render() {
-		const { backPath, complete, isMultisite, siteId, themeId, translate } = this.props;
-
+		const {
+			backPath,
+			complete,
+			isFetchingPurchases,
+			isJetpack,
+			isMultisite,
+			isOnAtomicPlan,
+			siteId,
+			themeId,
+			translate,
+		} = this.props;
+
+		const showUpgradeBanner = ! isFetchingPurchases && ! isOnAtomicPlan && ! isJetpack;
 		const { showEligibility } = this.state;
 
 		if ( isMultisite ) {
 			return this.renderNotAvailableForMultisite();
 		}
 
 		return (
-			<Main>
+			<Main className="theme-upload" wideLayout>
 				<PageViewTracker path="/themes/upload/:site" title="Themes > Install" />
+				<DocumentHead title={ translate( 'Install Theme' ) } />
+
+				<QuerySitePurchases siteId={ siteId } />
 				<QueryEligibility siteId={ siteId } />
 				<QueryActiveTheme siteId={ siteId } />
 				{ themeId && complete && <QueryCanonicalTheme siteId={ siteId } themeId={ themeId } /> }
 				<ThanksModal source="upload" />
 				<AutoLoadingHomepageModal source="upload" />
+
+				<FormattedHeader
+					brandFont
+					headerText={ translate( 'Themes' ) }
+					subHeaderText={ translate(
+						'If you have a theme in .zip format, you may install or update it by uploading it here. {{learnMoreLink}}Learn more{{/learnMoreLink}}.',
+						{
+							components: {
+								learnMoreLink: (
+									<InlineSupportLink
+										supportLink="https://wordpress.com/support/themes/uploading-setting-up-custom-themes/"
+										supportPostId={ 134784 }
+										showIcon={ false }
+									/>
+								),
+							},
+						}
+					) }
+					align="left"
+				/>
 				<HeaderCake backHref={ backPath }>{ translate( 'Install theme' ) }</HeaderCake>
+
+				{ showUpgradeBanner && this.renderUpgradeBanner() }
+
 				{ showEligibility && (
 					<EligibilityWarnings backUrl={ backPath } onProceed={ this.onProceedClick } />
 				) }
-				{ ! showEligibility && this.renderUploadCard() }
+
+				{ this.renderUploadCard() }
```

Comment:
We show this all the time, just using an overlay gradient (`FeatureExample`) when not enabled.